src/bench/bench.cpp

```diff
@@ -111,15 +111,25 @@ void BenchRunner::RunAll(const Args& args)
         std::cout << "Running with -sanity-check option, output is being suppressed as benchmark results will be useless." << std::endl;
     }
 
-    // Load inner test setup args
-    g_bench_command_line_args = [&args]() {
-        std::vector<const char*> ret;
-        ret.reserve(args.setup_args.size());
-        for (const auto& arg : args.setup_args) ret.emplace_back(arg.c_str());
-        return ret;
-    };
+    bool is_thread_scaling = args.scale_threads && (args.regex_filter != ".*");
+    if (args.scale_threads && !is_thread_scaling) {
+        std::cout << "Warning: -scale-threads requires -filter to be set. Running with default thread count." << std::endl;
+    }
+
+    std::vector<int> thread_scales;
```

Comment:
What's the purpose of a vector over a range? Are we planning on testing threads `2,4,8,16` instead of `1,2,3,4,5,6...`?
If not, consider making it a range, something like:
```C++
constexpr int DEFAULT_MAX_BENCH_THREADS{16};
const auto [start_threads, end_threads]{is_thread_scaling ? 
    std::pair{1, DEFAULT_MAX_BENCH_THREADS} : std::pair{0, 0}};
```
and maybe iterate as
```C++
for (int threads{start_threads}; threads <= end_threads; ++threads) {
    ...
}
```